components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -104,10 +107,57 @@ function Main() {
     }
   }
 
+  let inputFocused: boolean = false
+  let viewPortWithoutKeyboard = 0
+  let keyboardSize = 0
+
+  window.addEventListener('resize', () => {
+    if (!context.isMobile || !inputFocused || window.visualViewport == null) {
+      return
+    }
+    let viewPortWithKeyboard = window.visualViewport.height
+    if (headerElement == null || conversationContentElement == null ||
+      viewPortWithKeyboard === 0 || viewPortWithoutKeyboard === 0) {
+      return
+    }
+    if (keyboardSize === 0 ||
+        keyboardSize < viewPortWithoutKeyboard - viewPortWithKeyboard) {
+      keyboardSize = viewPortWithoutKeyboard - viewPortWithKeyboard
+    }
+    let mountPoint: HTMLElement | null = document.getElementById('mountPoint')
+    if (mountPoint !== undefined && mountPoint !== null) {
+      if (mountPoint.clientHeight >=
+          (headerElement.clientHeight + conversationContentElement.clientHeight)
+          * 2) {
+        let percent = viewPortWithKeyboard * 100 / viewPortWithoutKeyboard
+        mountPoint.style.height = `${percent}%`
+      } else if (keyboardSize > viewPortWithoutKeyboard - viewPortWithKeyboard) {
+        mountPoint.style.height = '100%'
+      }
+    }
+  })
```

Comment:
This should be wrapped in a `useEffect` which removes the listener - as is, we'll be adding a new listener every time the component rerenders (which is every time anything  in the context changes).

```tsx
useEffect(() => {
    const handler = () => { ... }
    window.addEventListener('resize', handler)
    return () => {
        window.removeEventListener('resize', handler)
    }
}, [])